debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/RomClassWalker.java

```diff
@@ -983,6 +989,21 @@ void permittedSubclassAttributeDo(U32Pointer attribute) throws CorruptDataExcept
 		classWalkerCallback.addSection(clazz, attributeStart, attribute.getAddress() - attributeStart.getAddress(), "permittedSubclass", true);
 	}
 
+	void preloadAttributeDo(U32Pointer attribute) throws CorruptDataException
+	{
+		if (attribute.isNull()) {
+			return;
+		}
+		U32Pointer attributeStart = attribute;
+		classWalkerCallback.addSlot(clazz, SlotType.J9_U32, attribute, "numberPreloadClasses");
+		for (int i = 0, numPreloadClasses = attribute.at(0).intValue(); i < numPreloadClasses; i++) {
+			attribute = attribute.add(1);
+			classWalkerCallback.addSlot(clazz, SlotType.J9_ROM_UTF8, attribute, "preloadClassName");
+		}
+		attribute = attribute.add(1);
```

Comment:
My suggestion was missing this line; perhaps following the pattern of other methods here makes more sense:
```java
		attribute = attribute.add(1);
		for (int i = 0, numPreloadClasses = attribute.at(0).intValue(); i < numPreloadClasses; i++) {
			classWalkerCallback.addSlot(clazz, SlotType.J9_ROM_UTF8, attribute, "preloadClassName");
			attribute = attribute.add(1);
		}
```

Reply:
I tried that first but `attribute.add` changes the address of attribute which will make numPreloadClasses incorrect.

Reply:
Sorry, you're right. Let's leave this as it is now.